examples/pxScene2d/src/pxResource.cpp

```diff
@@ -366,7 +366,9 @@ void rtImageResource::prepare()
 
 void rtImageResource::setTextureData(pxOffscreen& imageOffscreen, const char* data, const size_t dataSize)
 {
+  mTextureMutex.lock();
```

Comment:
Is this lock needed?  I think the code prevents mDownloadedTexture from being accessed by multiple threads at the same time without a mutex.

Reply:
It's a fix for #1257 - please have a look at the second frame in the two first backtraces:

(1)
>#2 rtImageResource::setTextureData(pxOffscreen&, char const*, unsigned long) /home/runner/pxCore/examples/pxScene2d/src/pxResource.cpp:369 (pxscene+0x476a9d)

(2)
>#2 rtImageResource::getTexture(bool) /home/runner/pxCore/examples/pxScene2d/src/pxResource.cpp:330 (pxscene+0x476811)

both accessing `mDownloadedTexture` but only `getTexture()`  in (2) is using `mTextureMutex` while `setTextureData()` doesn't seem to use it.